scripts/sct_utils.py

```diff
@@ -206,6 +206,99 @@ def run(cmd, verbose=1, error_exit='error', raise_exception=False):
         return status_output, output_final[0:-1]
 
 
+def check_exe(name):
+    """
+    Ensure that a program exists
+    :type name: string
+    :param name: name or path to program
+    :return: path of the program or None
+    """
+    def is_exe(fpath):
+        return os.path.isfile(fpath) and os.access(fpath, os.X_OK)
+
+    fpath, fname = os.path.split(name)
+    if fpath and is_exe(name):
+        return fpath
+    else:
+        for path in os.environ["PATH"].split(os.pathsep):
+            path = path.strip('"')
+            exe_file = os.path.join(path, name)
+            if is_exe(exe_file):
+                return exe_file
+
+    return None
+
+
+def display_viewer(l_file, l_colormap=[], l_minmax=[], l_opacity=[], mode='', verbose=1):
+    """
+    Print the syntax to open a viewer and display images for QC. To use default values, enter empty string: ''
+    Parameters
+    ----------
+    l_file [list:string]: list of NIFTI file names
+    l_colormap [list:string]: list of colormaps associated with each file. Available colour maps: blue, blue-lightblue, cool, copper, cortical, green, greyscale, hot, hsv, pink, random, red, red-yellow, render1, render1t, render2, render2t, render3, retino, subcortical, yellow. Default=greyscale.
+    l_minmax [list:string]: list of min,max brightness scale associated with each file. Separate with comma.
+    l_opacity [list:string]: list of opacity associated with each file. Between 0 and 1.
+
+    Returns
+    -------
+    None
+
+    Example
+    -------
+    sct.display_viewer([file1, file2, file3])
+    sct.display_viewer([file1, file2], l_colormap=['gray', 'red'], l_minmax=['', '0,1'], l_opacity=['', '0.7'])
+    """
+    list_viewer = ['fslview', 'fslview_deprecated', 'fsleyes']  # list of known viewers. Can add more.
+    dict_fslview = {'gray': 'Greyscale', 'red-yellow': 'Red-Yellow', 'blue-lightblue': 'Blue-Lightblue', 'red': 'Red', 'random': 'Random-Rainbow'}
+    dict_fsleyes = {'gray': 'greyscale', 'red-yellow': 'red-yellow', 'blue-lightblue': 'blue-lightblue', 'red': 'red', 'random': 'random'}
+    selected_viewer = None
+
+    # find viewer
+    for viewer in list_viewer:
+        if check_exe(viewer):
+            selected_viewer = viewer
+            break
+    if selected_viewer == None:
+        # exit without error
+        return
+
+    # loop across files and build syntax
+    cmd = viewer
```

Comment:
Strongly advise to construct a command-line with a list, not a string (I know the rest of the code doesn't do that, but this will have to change some day).
And then use `subprocess.Popen()` without shell.

Reply:
if i understand correctly, you suggest to use subprocess to run the syntax generated by `display_syntax_viewer()`. However in that case the goal of this function is to conveniently display the syntax on the Terminal window, not to run it.